contracts/contracts/staking/SingleAssetStaking.sol

```diff
@@ -191,6 +191,17 @@ contract SingleAssetStaking is Initializable, Governable {
         emit Staked(staker, amount);
     }
 
+    function _stakeWithChecks(address staker, uint256 amount, uint256 duration) internal {
+        require(amount > 0, "Cannot stake 0");
+
+        uint240 rewardRate = _findDurationRate(duration);
+        require(rewardRate > 0, "Invalid duration"); // we couldn't find the rate that correspond to the passed duration
+
+        _stake(msg.sender, USER_STAKE_TYPE, duration, rewardRate, amount);
```

Comment:
this should be staker rather than msg.sender

Reply:
great catch, thanks!